tasks/jsonpretty.js

```diff
@@ -5,63 +5,69 @@
  * Copyright (c) 2014 Ryan Burgess
  * Licensed under the MIT license.
  */
-
 'use strict';
 
-module.exports = function (grunt) {
-  grunt.registerTask('json-pretty', function(){
-    var fs = require('fs'),
-      options,
-      fileContent,
-      fileFull,
-      content,
-      minFile,
-      minContent,
-      minFileContent,
-      indent,
-      successful = 0,
-      failed = 0;
+module.exports = function(grunt) {
+    grunt.registerTask('json-pretty', function() {
+        var fs = require('fs'),
+            options,
+            directories = [],
+            json,
+            successful = 0,
+            failed = 0;
 
-    options = this.options({
-      indent: 2,
-      files: '/content/',
-      minify: false,
-      append: '.min'
-    });
+        options = this.options({
+            indent: 2,
+            files: '/content/',
+            minify: false,
+            append: '.min',
+            cleanup: false
+        });
+
+        //set directory(ies)
+        if (Array.isArray(options.src || options.files)) directories = options.src || options.files;
+        else directories.push(options.src || options.files);
 
-    // Get list of files depending on the file directory
-    grunt.file.expand({
-      filter: 'isFile',
-        cwd: options.files // Change this reference to your directory
-      }, 
-      ['**/*.json']).forEach(function(file){
+        //prettify & minify function
+        function pretty(directory) {
+            return function(file) {
+                try {
+                    //set file
+                    file = directory + file;
+                    //read the file and parse
+                    json = JSON.parse(fs.readFileSync(file));
+                    successful++;
 
-        try {
-          fileFull = options.files + file;
-          fileContent = fs.readFileSync(fileFull);
-          content = JSON.parse(fileContent);
-          successful++;
+                    fs.writeFileSync(file, JSON.stringify(json, false, options.indent));
 
-          //Serialize as JSON and Write it to a file
-          fs.writeFileSync(fileFull, JSON.stringify(content, false, options.indent));
-          if(options.minify !== false){
-        
-            if(fileFull.indexOf(options.append+'.json') > -1){
-              minContent = fs.readFileSync(minFile);
```

Comment:
I wasn't too sure what exactly `minFile` was here because it doesn't seem to be assigned (basically `undefined`) until after the if statement. So interpreted that as "if the file does contain `.min.js`, then read the minified file else read the un-minified file". Maybe I should have used `_file` instead of `minified`.